.github/workflows/build-ios.yaml

```diff
@@ -0,0 +1,122 @@
+name: Build for iOS
+
+on:
+  workflow_call:
+    inputs:
+      deploy:
+        description: "Whether or not to deploy the app to the App Store"
+        required: true
+        type: boolean
```

Comment:
question (non-blocking): Looks like this is only used in deploy flows as of this PR which makes sense. Appreciate it being controlled with an input var, but wondering if you envision this being used outside deploy flows?

Reply:
Seems like we don't envision this being used outside deploy flows, but keeping this argument around makes this workflow more similar to the build-android workflow, and it doesn't really add a meaningful amount of complexity, so I'm inclined to leave it in anyway.

Reply:
Yeah I'm cool with leaving it in place!